lib/spack/spack/cmd/buildcache.py

```diff
@@ -99,6 +115,68 @@ def setup_parser(subparser):
                         help="force new download of keys")
     dlkeys.set_defaults(func=getkeys)
 
+    # Check if binaries need to be rebuilt on remote mirror
+    check = subparsers.add_parser('check', help=check_binaries.__doc__)
+    check.add_argument(
+        '-m', '--mirror-url', default=None,
+        help='Override any configured mirrors with this mirror url')
+
+    check.add_argument(
+        '-o', '--output-file', default=None,
+        help='File where rebuild info should be written')
+
+    # used to construct scope arguments below
+    scopes = spack.config.scopes()
+    scopes_metavar = spack.config.scopes_metavar
+
+    check.add_argument(
+        '--scope', choices=scopes, metavar=scopes_metavar,
+        default=spack.cmd.default_modify_scope(),
+        help="configuration scope containing mirrors to check")
+
+    check.add_argument(
+        '-s', '--spec', default=None,
+        help='Check single spec instead of building from release specs file')
+
+    check.add_argument(
+        '-n', '--no-index', action='store_true', default=False,
+        help='Do not use buildcache index, instead retrieve .spec.yaml files')
+    check.set_defaults(func=check_binaries)
+
+    # Download tarball and spec.yaml
+    dltarball = subparsers.add_parser('download', help=get_tarball.__doc__)
+    dltarball.add_argument(
+        '-s', '--spec', default=None,
+        help="Download built tarball for spec from mirror")
+    dltarball.add_argument(
+        '-p', '--path', default=None,
+        help="Path to directory where tarball should be downloaded")
+    dltarball.set_defaults(func=get_tarball)
+
+    # Get buildcache name
+    getname = subparsers.add_parser('get-name',
+                                    help=get_buildcache_name.__doc__)
+    getname.add_argument(
+        '-s', '--spec', default=None,
+        help='Spec for which buildcache name is desired')
+    getname.set_defaults(func=get_buildcache_name)
+
+    # Put buildcache entry somewhere (file system or s3)
+    put = subparsers.add_parser('put', help=put_buildcache.__doc__)
+    put.add_argument('-d', '--directory', default='.',
+                     help="Location of mirror directory, similar to 'create'")
+    put.add_argument('-s', '--spec', default=None,
+                     help="Spec indicating which binary to put")
+    put.add_argument('-m', '--mirror-name', default=None,
+                     help="Name of mirror where binary should be pushed")
+    put.add_argument('-p', '--profile', default=None,
+                     help="Name of AWS profile")
+    put.add_argument(
+        '--scope', choices=scopes, metavar=scopes_metavar,
+        default=spack.cmd.default_modify_scope(),
+        help="configuration scope giving possible mirrors")
```

Comment:
...and here.

Whenever running any command hits this code path.  In fact, `spack help` is enough to hit this error.

Reply:
That's pretty weird, do you not see that method defined around line 46 of `lib/spack/spack/cmd/__init__.py`?

Reply:
uuhhh HA!  Yes, of course!  I was just testing you!  I certainly didn't do anything silly on my end.  Nothing to see here...

Reply:
So, just FYI, what I did was not completely crazy (I tried merging this into a temporary copy of my docker PR #9329).  It looks like there has been (at least) one relevant [commit](https://github.com/spack/spack/commit/6f5a68a58dec07c1e7699513d373b2e8a6fcd77f) made since you branched off, here.

In any case, it might be time for a rebase.